### ISSUE TITLE
TypeError: me.error is not a function

### ISSUE THREAD

**Issue by qq7te** [tootsuite/mastodon#600]
Hello, I have to admit I'm new to this, but there seems to be a problem with one of the libraries, potentially outside of the work done on mastodon. Something must have changed in one of the underlying libraries. 

I'm hoping that someone here will be able to quickly notice what the problem is, and either fix it or recommend a workaround
I have attached the final output from my dun of **docker-compose build**

yarn install v0.19.1
warning mastodon: No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
warning There appears to be trouble with your network connection. Retrying...
warning There appears to be trouble with your network connection. Retrying...
/usr/lib/node_modules/yarn/node_modules/tar/lib/extended-header.js:138
  me.error(msg)
     ^

TypeError: me.error is not a function
    at error (/usr/lib/node_modules/yarn/node_modules/tar/lib/extended-header.js:138:6)
    at ExtendedHeader.parse [as _parse] (/usr/lib/node_modules/yarn/node_modules/tar/lib/extended-header.js:89:11)
    at emitOne (events.js:77:13)
    at ExtendedHeader.emit (events.js:169:7)
    at ExtendedHeader.Entry._read (/usr/lib/node_modules/yarn/node_modules/tar/lib/entry.js:112:10)
    at ExtendedHeader.Entry.write (/usr/lib/node_modules/yarn/node_modules/tar/lib/entry.js:69:8)
    at Extract.Parse._process (/usr/lib/node_modules/yarn/node_modules/tar/lib/parse.js:105:29)
    at BlockStream.<anonymous> (/usr/lib/node_modules/yarn/node_modules/tar/lib/parse.js:47:8)
    at emitOne (events.js:77:13)
    at BlockStream.emit (events.js:169:7)
ERROR: Service 'sidekiq' failed to build: The command '/bin/sh -c yarn' returned a non-zero code: 1

**Comment by Gargron** [comment]
I'm running `docker-compose build` on my machine right now to see if I encounter something similiar (the builds are supposed to be reproducible, after all). If it doesn't throw that error then perhaps it was a network failure like the beginning of the error message signified.

**Comment by Gargron** [comment]
I think it might be a yarn or NPM availability issue, I got a random EOF, then run it again, then it ran fine. Just try again.

**Comment by qq7te** [comment]
Yes, I got a ton of those as well... it's been happening for the last few days. I was assuming that was a problem with my network, but I'm pleased to see that it's not just me. 

Anyway... the EOF problem is not the problem. I have re-tried a million times as well (I wouldn't submit something at the first failure). 
The problem is that if you get through with no EOF, then you end up with the javascript problem I pointed out. 
Were you able to rebuild the whole thing without issues?
